lenses/struct-lens.rkt

```diff
@@ -0,0 +1,31 @@
+#lang racket/base
+
+(provide struct-lens)
+
+(require racket/local
+         syntax/parse/define
+         alexis/util/struct
+         "main.rkt"
+         (for-syntax racket/base
+                     syntax/parse
+                     racket/syntax
+                     ))
+(module+ test
+  (require rackunit fancy-app (only-in lenses lens-transform*)))
+
+(define-simple-macro (struct-lens s:id fld:id)
+  #:with s-fld      (format-id #'s "~a-~a" #'s #'fld #:source #'fld)
+  #:with s-fld-set  (format-id #'s "~a-~a-set" #'s #'fld #:source #'fld)
+  (local [(define-struct-updaters s)]
```

Comment:
It's very neat, but is there a particular reason that it needs to be a top-level/module definitions context instead of an internal definitions context (for instance as introduced with `(let () ...)`)